File: kubwaannotations/src/main/java/com/wesleyjasonelliott/kubwa/annotation/MobileNumber.java
package com.wesleyjasonelliott.kubwa.annotation;

import com.wesleyjasonelliott.kubwa.rule.MobileNumberRule;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by wesley on 2016/07/28.
 */
@ValidateUsing(MobileNumberRule.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface MobileNumber {
    int errorMessage();
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/annotation/MobileNumber.kt
package com.wesleyelliott.kubwa.annotation

import com.wesleyelliott.kubwa.rule.MobileNumberRule

/**
 * Created by wesley on 2016/07/28.
 */
@ValidateUsing(MobileNumberRule::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CONSTRUCTOR,
    AnnotationTarget.VALUE_PARAMETER,
    AnnotationTarget.CLASS,
    AnnotationTarget.FILE
)
annotation class MobileNumber(val errorMessage: Int, val name: String = "mobileError") {

    @Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
    @Retention(AnnotationRetention.RUNTIME)
    @MustBeDocumented
    annotation class List(vararg val value: MobileNumber)
}
